### PR TITLE
Metric MDS

### PR DESCRIPTION
**Pull request by wildart**
Implementation of weighted (non)metric MDS with transformed proximities.

### CODE REVIEW

File: src/mmds.jl [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+```
+
+where `D` is a symmetric matrix `D` of distances between points.
+
+In addition, the `metric` parameter specifies type of MDS, By default, it is assigned with `nothing` value
+which results in performing *metric MDS* with dissimilarities calculated as Euclidian distances.
+
+An arbitrary transformation function can be provided to `metric` parameter to
+perform metric MDS with transformed proximities. The function has to accept two parameters,
+a vector of proximities and a vector of distances, in order to calculate disparities
```
Comment (timholy) [inline src/mmds.jl:26]: What do the vectors represent? I.e., if `p` is a proximity vector, what is the meaning of `p[i]`? Likewise, what is the meaning of `d[i]`?

File: src/common.jl [hunk 2]
```diff
@@ -120,3 +120,14 @@ function calcscattermat(Z::DenseMatrix)
...
 end
 
+
+# distance
+"""
+    L2distance(X)
+
+Calculate a symmetric Euclidean (L2) distance matrix.
+"""
+L2distance(X::AbstractMatrix{T}) where {T<:Real} = L2distance!(zeros(T,size(X,2),size(X,2)), X)
```
Comment (timholy) [inline src/common.jl:130]: Presumably we could use `Matrix{T}(undef, size(X, 2), size(X, 3))` but the performance gain is unlikely to be observable in practice.

**Review by timholy — COMMENTED** [review]
Sorry I didn't notice your review request earlier! Belated review below. Thanks for adding this, it looks like it may be useful!

File: src/mmds.jl (hunk 1 above)
Comment (wildart) [reply src/mmds.jl:26]: Proximity and distance vectors are compact (1D) representations of upper triangular part of proximity and distance matrices correspondingly.

**Pull request merged**